python/tests/api/logger/test_rolling.py

```diff
@@ -265,6 +268,48 @@ def test_rolling_row_messages_with_segments(tmp_path: Any) -> None:
     assert rolling_callback.call_count == 2
 
 
+@pytest.mark.load
+def test_rolling_logger_load_test(tmp_path: Any) -> None:
+    import gc
+    import tracemalloc
+
+    num_messages = 10
+    messages = [{"col1": i % 2, "col2": i * i * 1.2, "col3": "a"} for i in range(num_messages)]
+    tracemalloc.start()
+    # Here we create an aggressively rolling logger to try and test for memory pressure related to
+    # long running rolling logger instances over time. Don't do this in actual integrations.
+    rolling_logger = why.logger(
+        mode="rolling",
+        interval=1,
+        when="S",
+        base_name="test_base_name",
+    )
+    rolling_logger.append_writer("local", base_dir=tmp_path)
+    # parameterize the load test, 10,000 iterations with 10 messages per loop -> 100k log calls
+    test_iterations = 10000
+
+    def loop_test(rolling_logger, messages, test_iterations):
+        for i in range(test_iterations):
+            if i % 1000 == 0:
+                print(f"Iteration {i} out of {test_iterations}")
+            for message in messages:
+                rolling_logger.log(message)
+
+    loop_test(rolling_logger, messages, test_iterations)
+    gc.collect()
+    snapshot = tracemalloc.take_snapshot()
+    top_stats = snapshot.statistics("lineno")
+
+    TEST_LOGGER.info("Top memory-consuming lines:")
+    for stat in top_stats[:5]:
+        TEST_LOGGER.info(stat)
+
+    TEST_LOGGER.info(f"load test with {test_iterations} iterations each using {num_messages}")
+
+    # after explicitly calling close on the logger, we trigger one more flush which has two segments and expect two callbacks
```

Comment:
Can we assert anything?

Reply:
Let's do the assertion in a follow up. I added a comment on what to look for to help anyone running this test know what to look for.